packages/compartment-mapper/test/search.test.js

```diff
@@ -0,0 +1,56 @@
+import fs from 'fs';
+import crypto from 'crypto';
+import url from 'url';
+import test from 'ava';
+import { search } from '../src/search.js';
+import { makeReadPowers } from '../src/node-powers.js';
+
+test('search should find own package.json with read power', async t => {
+  const readPowers = makeReadPowers({
+    fs,
+    crypto,
+    url,
+  });
+  const { read } = readPowers;
+  const { packageDescriptorLocation: found } = await search(
+    read,
+    import.meta.url,
+  );
+  const sought = new URL('../package.json', import.meta.url).href;
+  t.is(found, sought);
+});
+
+test('search should find own package.json with read powers (plural)', async t => {
+  const readPowers = makeReadPowers({
+    fs,
+    crypto,
+    url,
+  });
+  const { packageDescriptorLocation: found } = await search(
+    readPowers,
+    import.meta.url,
+  );
+  const sought = new URL('../package.json', import.meta.url).href;
+  t.is(found, sought);
+});
+
+test('search should fail to find package.json at system root with read power', async t => {
+  const readPowers = makeReadPowers({
+    fs,
+    crypto,
+    url,
+  });
+  const { read } = readPowers;
+  const root = new URL('/', import.meta.url);
+  await t.throwsAsync(() => search(read, root));
+});
```

Comment:
Using the root directory makes me uncomfortable both in terms of accessing the filesystem outside of the endo checkout, as well as not being robust if somebody actually has a `/package.json`.  Can you instead fake the `fs` power provided to `makeReadPowers({ fs })`?